src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -1380,15 +1385,42 @@ def test_parse_array_dimensions_attributes():
     reader = FortranStringReader("dimension(3,5)")
     fparser2spec = Dimension_Attr_Spec(reader)
     shape = Fparser2Reader._parse_dimensions(fparser2spec, sym_table)
-    assert shape == [3, 5]
+    assert len(shape) == 2
+    assert shape[0].value == "3"
+    assert shape[1].value == "5"
 
     sym_table.add(DataSymbol('var1', INTEGER_TYPE))
+    sym_table.add(DataSymbol('var1_upper', INTEGER_TYPE))
+
     reader = FortranStringReader("dimension(var1)")
     fparser2spec = Dimension_Attr_Spec(reader)
     shape = Fparser2Reader._parse_dimensions(fparser2spec, sym_table)
     assert len(shape) == 1
     assert shape[0].symbol == sym_table.lookup('var1')
 
+    reader = FortranStringReader("dimension(0:3,var1)")
+    fparser2spec = Dimension_Attr_Spec(reader)
+    shape = Fparser2Reader._parse_dimensions(fparser2spec, sym_table)
+    # First dim is specified with both lower and upper bounds so should
+    # have a tuple
+    assert isinstance(shape[0], tuple)
+    assert len(shape[0]) == 2
+    assert shape[0][0].value == "0"
+    assert shape[0][1].value == "3"
```

Comment:
Ah. That revealed that the docstring for _parse_dimensions in the fparser2 frontend was wrong. I've altered the implementation now so that it *always* returns a 2-tuple for every dimension that has a defined extent. I've updated the docstring and tests appropriately.